src/views/tenants/tenants.jsx

```diff
@@ -1,140 +1,273 @@
+  const [selectedTenants, setSelectedTenants] = useState([]);
+  const [nonSelectableRows, setNonSelectableRows] = useState([])
+  const [showArchived, setShowArchived] = useState(false);
+  const [checkboxRenderCount, setCheckboxRenderCount] = useState(0)
+  const [showArchiveModal, setShowArchiveModal] = useState(false)
+
+  useMountEffect(() => {
+    fetchAllTenants();
+  })
+
+  const fetchAllTenants = () => {
+    let allProperties;
+
+    axios.get(`/api/properties`, makeAuthHeaders(userContext))
+      .then(propertyResponse => {
+        const { data: { properties } } = propertyResponse;
+        allProperties = properties;
       })
+      .then(() =>
+        axios.get("/api/tenants", makeAuthHeaders(userContext))
+      )
       .then((response) => {
-        const tenants = response.data.tenants;
-        this.setState({ tenants });
+        const { data: { tenants } } = response;
+        const tenantRows = formatTenantData(tenants)
+        const tenantsWithProperties = addPropertyNames(tenantRows, allProperties)
+        const sortedTenants = sortTenantData(tenantsWithProperties)
+        setAllTenants(sortedTenants);
+        setSearchedTenants(sortedTenants);
+        setDisplayTenants(getDisplayTenants(sortedTenants, showHoused, showArchived));
+
+        const archivedTenants = sortedTenants.filter(tenant => tenant.archived);
+        setNonSelectableRows(archivedTenants.map(archivedTenant => archivedTenant.id))
+
       })
       .catch((error) => {
-        Toast(error.message, "error");
-        console.log(error);
+        Toast(error.message, "error")
       });
-  };
-
-  render() {
-    return (
-      <UserContext.Consumer>
-        {session => {
-          this.context = session;
-          return (
-            <div className='main-container'>
-              <div>
-                <div className="section-header">
-                  <h2 className="page-title">Tenants</h2>
-                  <Link className="button is-primary is-rounded ml-4" to="/add/tenant">+ ADD NEW</Link>
-                </div>
-                <div className="search-section">
-                  <Search placeholderMessage="Search tenants by name, property, or JOIN staff" />
-                </div>
-                <div className="properties-list">
-                  <BootstrapTable
-                    keyField='id'
-                    data={this.state.tenants}
-                    columns={columns}
-                    selectRow={selectRow}
-                    bootstrap4={true}
-                    headerClasses="table-header"
-                    classes="table-responsive"
-                  />
-                </div>
-              </div>
-            </div>
-          );
-        }}
-      </UserContext.Consumer>
-    );
   }
-}
+
+  const handleToggleHoused = () => {
+    const newShowHoused = !showHoused;
+    setShowHoused(newShowHoused)
+    setDisplayTenants(getDisplayTenants(allTenants, !showHoused, showArchived));
```

Comment:
If I have something in the search bar, and then toggle either the Unhoused or Archived switches, my search filter is unapplied even though it's still in the search bar. We might want to switch this handler function and the handleToggleArchived to setDisplayTenants based on the current searched tenants instead of reverting back to the full list. Either that or we clear the search box since the filter is not getting applied anymore.

![image](https://user-images.githubusercontent.com/8600792/112732633-4bd1b400-8ef8-11eb-8ea5-e73c7e5f36d5.png)

Reply:
Thank you for catching this!  I fixed it by making the list of tenants it pulls from conditional on whether search is active.  I believe its resolved, but let me know if I missed anything.